tests/unit/lms/services/canvas_test.py

```diff
@@ -303,18 +197,27 @@ def CanvasService(self, patch):
 
 @pytest.fixture
 def canvas_service(canvas_api_client, file_service):
-    canvas_service = CanvasService(canvas_api_client, file_service)
-    canvas_service.api.list_files.return_value = [
-        {"id": 1, "display_name": "File 1", "size": 1024},
-        {
-            "id": 2,
-            "display_name": "File 2",
-            "size": 2048,
-        },
-        {
-            "id": 3,
-            "display_name": "File 3",
-            "size": 3072,
-        },
-    ]
-    return canvas_service
+    return CanvasService(canvas_api_client, file_service)
+
+
+@pytest.fixture
+def file_record(file_service):
+    file_record = factories.File()
+    file_service.get.return_value = file_record
+
+    return file_record
+
+
+@pytest.fixture
+def matching_file_in_course(file_record, canvas_service):
+    # Create a match which is tailored to the file record
+    matching_file_in_course = {
+        "id": "new_file",
+        "display_name": file_record.name,
+        "size": file_record.size,
+    }
```

Comment:
Driving the matches backwards this way makes it a bit easier to understand I think. Previously we were pulling items from the return values of `list_files` and working our way backwards.

I think this better reflects how we're thinking of it, which is there is a result in `list_files` that matches our historical record. Rather than our historical record being driven by the results obtained from the list today.